server/models/dataset_helpers/image_metadata.py

```diff
@@ -515,4 +518,7 @@ def addImageClinicalMetadata(image, data):
     # Add remaining data as unstructured metadata
     image['meta']['unstructured'].update(data)
 
-    return errors
+    # Report unrecognized fields when there are no errors
+    unrecognizedFields = six.viewkeys(data) if not errors else None
```

Comment:
I think a `list` would be simpler here, and adequate for now. It can just be empty if there are no `unrecognizedFields`.

This does bring up the point that we're mutating the `data` parameter in-place. I'm not sure if this is a good design, as we could make a copy of `data` before calling each of the `parser.run` on it. @msmolens What do you think?

Reply:
Now this function operates on a copy of `data`.